libraries/chain/controller.cpp

```diff
@@ -831,6 +833,35 @@ rpc::chain::get_account_nonce_response controller_impl::get_account_nonce( const
    return resp;
 }
 
+rpc::chain::invoke_system_call_response controller_impl::invoke_system_call( const rpc::chain::invoke_system_call_request& request )
+{
+   execution_context ctx( _vm_backend, intent::read_only );
+   ctx.push_frame( stack_frame {
+      .call_privilege = privilege::user_mode
+   } );
+
+   ctx.set_state_node( _db.get_head( _db.get_shared_lock() )->create_anonymous_node() );
+   ctx.reset_cache();
+
+   koinos::chain::execution_result res;
+   if (request.has_id())
+   {
+      res = ctx.system_call( request.id(), request.args() );
+   }
+   else
+   {
+      system_call_id val;
+      if (!system_call_id_Parse( request.name(), &val ))
+         KOINOS_THROW( unknown_system_call_exception, "Invalid system call name" );
```

Comment:
All `KOINOS_THROW` messages should be lowercase.

We should change `"Invalid system call name"` to `"unknown system call name"`. The system call name could be valid, yet it is unknown to the version of `koinos-proto` running the microservice.